articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+budicon: 448
+topics:
+  - quickstarts
+  - webapp
+  - login
+  - user profile
+  - logout
+  - nodejs
+contentType: tutorial
+useCase: quickstart
+github:
+  path: 01-Login
+---
+
+<!-- markdownlint-disable MD002 -->
+## Getting Started
+In this quickstart you will learn how to implement authentication in an Express application using the OIDC Compliant library [Express OIDC](https://github.com/auth0/express-openid-connect) authored and maintained by Auth0.
+
+::: warning
+Express OIDC is still in an [early access stage](/).
+:::
+
+### Install Dependencies
+
+```sh
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+## Configure Auth0
+### Configure Callback URLs
+
+A callback URL is a URL in your application where Auth0 redirects the user after they have authenticated. 
+
+The callback URL for your app must be whitelisted in the **Allowed Callback URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log in to the application and will get an error.
+
+::: note
+If you are testing your application locally, you should set the **Allowed Callback URL** to `http://localhost:3000/callback`
+:::
+
+### Configure Logout URLs
+
+A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter.
+
+The logout URL for your app must be whitelisted in the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log out from the application and will get an error.
+
+::: note
+If you are testing your application locally, the logout URL you need to whitelist in the **Allowed Logout URLs** field is `http://localhost:3000`.
+:::
+
+## Integrate Auth0
+### Auth0 Application Keys
+
+You will need to retrieve some details for your application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) section in the Auth0 dashboard.
+
+You need the following information:
+
+* **Domain**
+* **Client ID**
+* **Client Secret**
+
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
+- `express-session` - simple session middleware for Express (other session middleware can be used as a replacement).
+- `express-openid-connect` - an [OIDC](/protocols/oidc) compliant authentication middleware.
+
+::: note
+To learn more about using middleware with Express view the [official documentation](https://expressjs.com/en/guide/using-middleware.html)
+:::
+
+::: note
```

Comment:
This is a good note to have but ... we've got "secrets" in the code below. In other QSes, we say something like this but integrate our recommended library into the QS.

Reply:
User research was showing that users wanted to handle secret management at a different stage and outside of the quickstart.  This is more of a PSA than anything.

Reply:
Interesting ... so we're thinking that the environment stuff should be left out, besides a note like this? 

I think I'm biased because the whole "environment storage" was such a catch point for me early on in my dev experience. The big downside of including it in the QS is, of course, that this is duplicated across all QSes in this platform.

Reply:
The only reason for the note was because of sentiment shared internally that we need to signal to the user best practices in security, more specifically not to check secrets into the source code.  

Also discussed here https://github.com/auth0/docs/pull/8355/files/cd66335815abeb2f2bb399299c82f6408e5432cf#r333791381 possibly it is best to show the user how to manage secrets, but it does add an additional dependency and widens scope of what the user is doing in this tutorial